File: comandodevoz/app/src/main/java/com/example/comandodevoz/MainActivity.java
package com.example.comandodevoz;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button ComandoVoz;
    Button ComandoPorEscrito;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        ComandoVoz        = findViewById(R.id.ComandoVoz);
        ComandoPorEscrito = findViewById(R.id.ComandoPorEscrito);


        ComandoVoz.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent a = new Intent(MainActivity.this,MainActivity2.class);
                startActivity(a);
            }
        });

        ComandoPorEscrito.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent a = new Intent(MainActivity.this,MainActivity3.class);
                startActivity(a);
            }
        });

    }
}

File: kotlin/app/src/main/java/com/example/kotlin/MainActivity.kt
package com.example.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.speech.tts.TextToSpeech
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import java.util.*

class MainActivity : AppCompatActivity(), TextToSpeech.OnInitListener{




    var tts: TextToSpeech? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tts = TextToSpeech(this,this)

        findViewById<Button>(R.id.btnlose)
        findViewById<Button>(R.id.btnplay).setOnClickListener { speak() }





    }//fin onCreate

    private fun speak(){
        var message: String = findViewById<TextView>(R.id.etMessage).text.toString()


        if(message.isEmpty()){
            findViewById<TextView>(R.id.tvStatus).text = "INTRODUCE UN TEXTO"
            message = "POR FAVOR INTRODUCE UN TEXTO"
            //Toast.makeText(this,"poner un texto",Toast.LENGTH_LONG).show()
        }

        tts!!.speak(message,TextToSpeech.QUEUE_FLUSH,null,"")
    }

    override fun onInit(status: Int) {

        if (status == TextToSpeech.SUCCESS){
            findViewById<TextView>(R.id.tvStatus).text = "LISTO"
            tts!!.setLanguage(Locale("ES"))
        }else{
            findViewById<TextView>(R.id.tvStatus).text = "NO DISPOSIBLE"
        }

    }

    override fun onDestroy() {
        if (tts != null) {
            tts!!.stop()
            tts!!.shutdown()
        }
        super.onDestroy()
    }


}//fin MainActivity